### DISCUSSION TITLE
How to change the backgroundColor of the stage

### DISCUSSION THREAD

**Discussion by lykl** [pixijs/pixijs#9043] [Q&A] [answered]
Now it can only be set when the `Application` is created, so how to change it later?

**Comment by SuperSodaSea** [comment, marked as answer]
For PixiJS v7:
```ts
app.renderer.background.color = 0x123456;
```

For PixiJS v6:
```ts
app.renderer.backgroundColor = 0x123456;
```

**Reply by lykl** [reply]
Doesn't seem to work, and typescript throws an error

**Reply by SuperSodaSea** [reply]
If you are using PixiJS v6.x, you need to set `backgroundColor` instead.

**Reply by lykl** [reply]
> If you are using PixiJS v6.x, you need to set `backgroundColor` instead.

The version is 7.x, this doesn't work because `color` only accept hexadecimal number type, not hexadecimal string, but typescript errors still exist

**Reply by lykl** [reply]
Here is my code now:
```typescript
 /* @ts-ignore */
 this.app.renderer.background.color = parseInt(color.replace(/#/, ""), 16);
```

**Reply by lykl** [reply]
`IRenderer<ICanvas>` doesn't have prop named `background`

**Reply by SuperSodaSea** [reply]
Try this:
```ts
import type { Renderer } from '@pixi/core';

(this.app.renderer as Renderer).background.color` = ...;
```

**Reply by lykl** [reply]
> Try this:
> 
> ```ts
> import type { Renderer } from '@pixi/core';
> 
> (this.app.renderer as Renderer).background.color` = ...;
> ```

Cool, that's perfect.

**Reply by SuperSodaSea** [reply]
I will make PRs to solve the problems later (`background.color` accept string, `IRenderer.background`).

**Reply by lykl** [reply]
> I will make PRs to solve the problems later (`background.color` accept string, `IRenderer.background`).

That's nice, and I thank you on behalf of anyone who might have the same problem in the future.
